deploy/1_seed.ts

```diff
@@ -0,0 +1,16 @@
+import { Logger } from 'ethers/lib/utils';
+import { validateDeployment, configureDeploymentSettings, seedContracts } from '../utils/deploy';
+import { LogLevel } from '@ethersproject/logger';
+import { getContractsFromDeployments } from '../test/helpers/index';
+
+export const deploy: CustomHardhatDeployFunction = async (env) => {
+    const hre = env as unknown as CustomHardHatRuntimeEnvironment;
```

Comment:
let's just fix the `CustomHardhatDeployFunction`

```ts
  type OmitCallSignature<T> = { [K in keyof T]: T[K] } & (T extends new (
    ...args: infer R
  ) => infer S
    ? new (...args: R) => S
    : unknown);

  interface CustomHardhatDeployFunction
    extends OmitCallSignature<HardhatDeployFunction> {
    (env: CustomHardHatRuntimeEnvironment): Promise<void | boolean>;
  }
```

then you can just rename the parameter as `hre` and drop the casting all together (the below is also linted)

```ts
import { Logger } from 'ethers/lib/utils';
import { LogLevel } from '@ethersproject/logger';

import { seedContracts } from '../utils/deploy';
import { getContractsFromDeployments } from '../test/helpers/index';

export const deploy: CustomHardhatDeployFunction = async (env) => {
  const hre = env;
  Logger.setLogLevel(LogLevel.DEBUG);
  hre.log(`1_seed`);
  const contracts = await getContractsFromDeployments(hre);
  await seedContracts({ hre, contracts });
};

export default deploy;
deploy.tags = ['seed'];
deploy.runAtTheEnd = true;

```